cpp/src/copying/contiguous_split.cu

```diff
@@ -0,0 +1,550 @@
+      // 128k calls to this function.
+      //
+      // 1. 500 ms    : no valdity information.
+      // 2. 10,000 ms  : validify information, leaving UNKNOWN_NULL_COUNTS in place. the time difference 
+      //    here is in the null_count() recomputation that happens in column_device_view::create() and the time
+      //    spent allocating/reading from the device scalar to get the resulting null count
+      // 3. 3,600 ms  : validity information, faking 0 input null count,  allocating a device scalar on the spot, 
+      //    recomputing null count in the copy_in_place_kernel and reading it back.
+      // 4. 2,700 ms  : validity information, faking 0 input null count, keeping a global device scalar, 
+      //    recomputing null count in the copy_in_place_kernel and reading it back.
+      // 5. 500 ms    : validity information, faking 0 input null count, setting output null count to UNKNOWN_NULL_COUNT. the
+      //    implication here of course is that someone else might end up paying this price later on down the road.
+      //
+      // Summary : nothing super surprising.  anything that causes memory allocation or copying between host and device
+      //           is super slow and becomes extremely noticeable at scale. best bet here seems to be to go with case 5 and 
+      //           let someone else pay the cost for lazily evaluating null counts down the road.
+      //
+      //
+
+      // see performance note above about null counts.
+      column_view          in_hacked{  in.type(), in.size(), in.head<T>(), 
+                                       in.null_mask(), in.null_mask() == nullptr ? UNKNOWN_NULL_COUNT : 0,
+                                       in.offset() };
+      mutable_column_view  mcv{in.type(), in.size(), data, 
+                               validity, validity == nullptr ? UNKNOWN_NULL_COUNT : 0 };      
+      if(in.nullable()){         
+         // PERFORMANCE TEST 2, 3
+         // rmm::device_scalar<cudf::size_type> valid_count{0, 0, rmm::mr::get_default_resource()};
+         copy_in_place_kernel<block_size, T, true><<<grid.num_blocks, block_size, 0, 0>>>(
+                           *column_device_view::create(in_hacked), 
+                           *mutable_column_device_view::create(mcv)
+                           // PERFORMANCE TEST 2, 3, 4
+                           //, valid_count.data()
+                           );
+         // PERFORMANCE TEST 2, 3, 4
+         // mcv.set_null_count(in.size() - valid_count.value());                  
+      } else
+       {
+         copy_in_place_kernel<block_size, T, false><<<grid.num_blocks, block_size, 0, 0>>>(
+                           *column_device_view::create(in_hacked), 
+                           *mutable_column_device_view::create(mcv)
+                           // PERFORMANCE TEST 2, 3, 4
+                           /*, nullptr*/);
+      }
+      mcv.set_null_count(cudf::UNKNOWN_NULL_COUNT);
```

Comment:
This isn't doing anything. This modifies a function local object which then is destroyed.

Reply:
It is.  That mcv gets converted to a column_view and put on the output list.